src/patchkernel/patch_kernel.hpp

```diff
@@ -497,17 +497,22 @@ friend class PatchManager;
 	CellIterator addCell(ElementType type, const std::vector<long> &connectivity, long id = Element::NULL_ID);
 	CellIterator addCell(ElementType type, std::unique_ptr<long[]> &&connectStorage, long id = Element::NULL_ID);
 #if BITPIT_ENABLE_MPI==1
-	CellIterator addCell(const Cell &source, int owner, long id = Element::NULL_ID);
-	CellIterator addCell(Cell &&source, int owner, long id = Element::NULL_ID);
-	CellIterator addCell(ElementType type, int owner, long id = Element::NULL_ID);
-	CellIterator addCell(ElementType type, const std::vector<long> &connectivity, int owner, long id = Element::NULL_ID);
-	CellIterator addCell(ElementType type, std::unique_ptr<long[]> &&connectStorage, int owner, long id = Element::NULL_ID);
+	BITPIT_DEPRECATED(CellIterator addCell(const Cell &source, int owner, long id = Element::NULL_ID));
```

Comment:
Can we maintain all the deprecated methods with no haloLayer argument?  

They are useful overloading methods with default 0-value of haloLayer that I prefer to save for a couple of reasons:

- the library keeps its compatibility with the previous version and all the applications based on bitpit are saved (they are currently forcely using only the first layer of ghosts);
- these methods remains consistent with all the other methods with default value of halo size/layer, as for instance the constructor of SurfUnstructured/VolUnstructured where this default value exists.

Reply:
Just a note, but @andrea-iob you know because your ongoing review, please consider to rebase on #340 and to use messages to indicate alternative methods.

Mostly when multiple functions are deprecated all together, for a bitpit user can be useful to have a message directly during the compilation with the function that he has to use instead of to have to navigate inside the headers.

Reply:
> Can we maintain all the deprecated methods with no haloLayer argument?

Done. However, the two set of function have very similar signatures, this might generate problems.